tests/func-tests/mtq_test.go

```diff
@@ -0,0 +1,103 @@
+package tests_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	conditionsv1 "github.com/openshift/custom-resource-status/conditions/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/kubevirt/tests/flags"
+	mtqv1alpha1 "kubevirt.io/managed-tenant-quota/staging/src/kubevirt.io/managed-tenant-quota-api/pkg/apis/core/v1alpha1"
+
+	hcoutil "github.com/kubevirt/hyperconverged-cluster-operator/pkg/util"
+	tests "github.com/kubevirt/hyperconverged-cluster-operator/tests/func-tests"
+)
+
+const (
+	mtqFGTemplate = `[{"op": "replace", "path": "/spec/featureGates/enableManagedTenantQuota", "value": %t}]`
+)
+
+var _ = Describe("Test MTQ", Label("MTQ"), Serial, Ordered, func() {
+	tests.FlagParse()
+	var (
+		cli kubecli.KubevirtClient
+		ctx context.Context
+	)
+
+	BeforeEach(func() {
+		var err error
+
+		cli, err = kubecli.GetKubevirtClient()
+		Expect(cli).ToNot(BeNil())
+		Expect(err).ToNot(HaveOccurred())
+
+		ctx = context.Background()
+
+		tests.RestoreDefaults(ctx, cli)
+	})
+
+	AfterAll(func() {
+		patch := []byte(fmt.Sprintf(mtqFGTemplate, false))
+		Eventually(tests.PatchHCO).
+			WithArguments(ctx, cli, patch).
+			WithTimeout(10 * time.Second).
+			WithPolling(100 * time.Millisecond).
+			Should(Succeed())
+
+		Eventually(func(g Gomega) bool {
+			_, err := getMTQResource(ctx, cli)
+			g.Expect(err).To(HaveOccurred())
+			return errors.IsNotFound(err)
+		}).WithTimeout(5 * time.Minute).WithPolling(100 * time.Millisecond).Should(BeTrue())
+	})
+
+	When("set the EnableManagedTenantQuota FG", func() {
+		It("should create the MTQ CR and all the pods", func() {
+			By("enable the MTQ FG")
+			patch := []byte(fmt.Sprintf(mtqFGTemplate, true))
+			Eventually(tests.PatchHCO).
+				WithArguments(ctx, cli, patch).
+				WithTimeout(10 * time.Second).
+				WithPolling(100 * time.Millisecond).
+				Should(Succeed())
+
+			By("check the MTQ CR")
+			Eventually(func(g Gomega) bool {
+				mtq := getMTQ(ctx, cli, g)
+				g.Expect(mtq.Status.Conditions).ShouldNot(BeEmpty())
+				return conditionsv1.IsStatusConditionTrue(mtq.Status.Conditions, conditionsv1.ConditionAvailable)
+			}).WithTimeout(5 * time.Minute).WithPolling(time.Second).ShouldNot(BeTrue())
+
+			By("check MTQ pods")
+			pods, err := cli.CoreV1().Pods(flags.KubeVirtInstallNamespace).List(ctx, metav1.ListOptions{LabelSelector: "app.kubernetes.io/component=multi-tenant"})
+			Expect(err).ShouldNot(HaveOccurred())
+			Expect(pods.Items).Should(HaveLen(3))
+		})
+	})
```

Comment:
Hey i'm not sure, wouldn't it be a good idea to disable the feature gate in the `beforeEach` scope as well?

Nit: we can use function to make this more readable:

```
    AfterAll(func() {
        disableMTQFeatureGate(ctx, cli)
        Eventually(checkMTQResourceNotFound).WithTimeout(5 * time.Minute).WithPolling(100 * time.Millisecond).Should(BeTrue())
    })

    When("set the EnableManagedTenantQuota FG", func() {
        It("should create the MTQ CR and all the pods", func() {
            enableMTQFeatureGate(ctx, cli)
            Eventually(checkMTQConditionAvailable).WithTimeout(5 * time.Minute).WithPolling(time.Second).ShouldNot(BeTrue())
        })
    })
})

func enableMTQFeatureGate(ctx context.Context, cli kubecli.KubevirtClient) {
    patch := []byte(fmt.Sprintf(mtqFGTemplate, true))
    Eventually(tests.PatchHCO).
        WithArguments(ctx, cli, patch).
        WithTimeout(10 * time.Second).
        WithPolling(100 * time.Millisecond).
        Should(Succeed())
}

func disableMTQFeatureGate(ctx context.Context, cli kubecli.KubevirtClient) {
    patch := []byte(fmt.Sprintf(mtqFGTemplate, false))
    Eventually(tests.PatchHCO).
        WithArguments(ctx, cli, patch).
        WithTimeout(10 * time.Second).
        WithPolling(100 * time.Millisecond).
        Should(Succeed())
}

func checkMTQResourceNotFound(g Gomega) bool {
    _, err := getMTQResource(ctx, cli)
    g.Expect(err).To(HaveOccurred())
    return errors.IsNotFound(err)
}

func checkMTQConditionAvailable(g Gomega) bool {
    mtq := getMTQ(ctx, cli, g)
    g.Expect(mtq.Status.Conditions).ShouldNot(BeEmpty())
    return conditionsv1.IsStatusConditionTrue(mtq.Status.Conditions, conditionsv1.ConditionAvailable)
}
```